File: API/Helpers/UserParams.cs
using System;
using CloudinaryDotNet.Actions;

namespace API.Helpers;

public class UserParams: PaginationParams
{


    public string? CurrentUsername { get; set; } // Username of the current user
    public string? Gender { get; set; } //
    public int minAge { get; set; } = 18; // Minimum age
    public int MaxAge { get; set; } = 100; // Maximum age
    public string OrderBy { get; set; } = "lastActive"; // Default order by last active
}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
  pageNumber = 1;
  pageSize = 5;
  gender: string;
  minAge = 18;
  maxAge = 99;
  orderBy = 'lastActive';
  constructor (user: User | null) {
    this.gender = user?.gender === 'female' ? 'mail' : 'female';
  }
}
